lib/Arch/Sleigh/ControlFlowStructuring.cpp

```diff
@@ -0,0 +1,470 @@
+#include "ControlFlowStructuring.h"
+
+#include <glog/logging.h>
+
+namespace remill::sleigh {
+
+bool isVarnodeInConstantSpace(VarnodeData vnode) {
+  auto spc = vnode.getAddr().getSpace();
+  return spc->constant_space_index == spc->getIndex();
+}
+
+
+namespace {
+
+// variant casting taken c&ped.
+template <class... Args>
+struct variant_cast_proxy {
+  std::variant<Args...> v;
+
+  template <class... ToArgs>
+  operator std::variant<ToArgs...>() const {
+    return std::visit([](auto &&arg) -> std::variant<ToArgs...> { return arg; },
+                      v);
+  }
+};
+
+template <class... Args>
+auto variant_cast(const std::variant<Args...> &v)
+    -> variant_cast_proxy<Args...> {
+  return {v};
+}
+
+enum CoarseEffect { ABNORMAL, NORMAL };
```

Comment:
`class`?

Reply:
i dont get it? is this referring to CoarseFlow maybe?